src/react_native/utils.cljs

```diff
@@ -37,3 +40,24 @@
         props       (when (map? first-child) first-child)
         children    (if props (rest argv) argv)]
     [props children]))
+
+(defn convert-prop-value
+  "Based on `reagent.impl.template/kv-conv`.
+  Takes the prop map of a reagent component and returns a React-valid property JS Object
+  by transforming kebab-case keys -> camelCase, maps -> JS Objects and vectors -> arrays.
+
+  This version adds support to recursively transform properties inside vectors, to have a
+  more consistent developer experience in React Native."
+  [x]
+  (cond
+    (reagent.util/js-val? x) x
+    (reagent.util/named? x)  (name x)
+    (map? x)                 (reduce-kv template/kv-conv #js {} x)
+    (vector? x)              (transforms/map-array convert-prop-value x)
```

Comment:
@flexsurfer especifically this line is the one that allows the transformation from kebab-case keywords -> camelCase JS react props on vectors recursively.

The line above checks for `(map? x)`, that's why reagent recursively transforms maps nested, so this PR adds the check for `vector?` too and then recursively transforms the keys